compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt

```diff
@@ -49,12 +49,6 @@ inline val FirRegularClass.isAbstract get() = status.modality == Modality.ABSTRA
 inline val FirRegularClass.canHaveAbstractDeclaration: Boolean
     get() = isAbstract || isSealed || isEnumClass
 
-inline val FirRegularClass.isInner get() = status.isInner
```

Comment:
Why those functions should be allowed on FirMemberDeclaration? 
It is applicable only to classes

Reply:
Sorry for the delayed reply, I was out of office and I missed the email reminder.

In order to test these changes, I made changes to `FirRenderer` to make sure the `inline` flag is not loaded. I decided to render all the status flags as well for all declarations, even if they are not applicable, just to make sure no incorrect flags are loaded for any of them.

Reply:
In such a case, I recommend not to change those functions but instead check for status explicitly in the renderer
As those functions are part of a wide-used API

Reply:
Good idea! Done. I'm also rendering `fun` for completeness.

In my defense, there are other util functions there that aren't specific to the appropriate declaration type (e.g., `isTailRec` should only be for `FirSimpleFunction`, `isLateInit` for `FirProperty`) but I won't touch those right now.

Reply:
@semoro As discussed in our sync, I am not rendering `fun` for Java interfaces. Updated the PR.